modules/mini-games/tightspot.lua

```diff
@@ -0,0 +1,758 @@
+        local force = entity[3]
+        local minable = entity[4]
+        position.x = position.x + at_player * 500
+        if name == "market" then
+            markets[#markets + 1] = position
+        end
+        --[[
+        if name == "steel-chest" then
+            chests[#chests+1] = {entity,player.name}
+        end
+        ]]
+        local ent = variables["surface"].create_entity{name = name , position = position , force = force }
+        if name == "steel-chest" then
+            chests[#chests+1] = {ent,player.name}
+        end
+        ent.minable = minable
+    end
+    centers[player.name] = {}
+    centers[player.name].x = level.center.x
+    centers[player.name].y = level.center.y
+    centers[player.name].x =  centers[player.name].x + at_player * 500
+    variables.walls[player.name] = {}
+    local wal = variables["surface"].find_entities_filtered {name = "stone-wall", area= area}
+    for i,wall in ipairs(wal) do
+        local p = wall.position
+        variables.walls[player.name][p.x..','..p.y] = true
+    end
+    local center = centers[player.name]
+    player.teleport({center.x,center.y},level.surface)
+    game.print("tp")
+end
+
+
+local function level_save()
+    local level = variables.level
+    for x=level.area[1][1],level.area[2][1] do
+        for y = level.area[1][2],level.area[2][2] do
+            local tile = variables["surface"].get_tile(x,y)
+            local table = {
+                name = tile.name,
+                position = tile.position
+            }
+            save["tiles"][#save["tiles"]+1] = table
+        end
+    end
+
+    save["entity"] = variables["surface"].find_entities_filtered {area = level.area}
+    for i, entity in ipairs(save["entity"]) do
+        local name = entity.name
+        if name ~= "character" then
+            if name == "steel-chest" then
+                chests[#chests+1] = {entity,game.connected_players[1].name}
+            end
+            local position = entity.position
+            local force = entity.force
+            local minbale =  entity.minable
+            local table = {name,position,force,minbale}
+            if name == "market" then
+                markets[#markets + 1] = position
+            end
+            save["entity"][i] = table
+        else
+            if i == #save["entity"] then
+                save["entity"][i] = nil
+            else
+                name = save["entity"][#save["entity"]].name
+                local position = save["entity"][#save["entity"]].position
+                local force = save["entity"][i].force
+                local minbale =  save["entity"][i].minable
+                local table = {name,position,force,minbale}
+                if name == "market" then
+                    markets[#markets + 1] = position
+                end
+                if name == "steel-chest" then
+                    chests[#chests+1] = {entity,game.connected_players[1].name}
+                end
```

Comment:
Becuse of the previous if statement you know that this code is only ran for player characters, so what is the point of these checks?

Reply:
no this is the last element i am chanching the player index to the last so its goan so this is some random entety which is the last in the array